packages/edit-site/src/components/page-pages/utils.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+import { privateApis as routerPrivateApis } from '@wordpress/router';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+
+const { useLocation } = unlock( routerPrivateApis );
+
+// Used on PagePages and PageTemplates.
+export function usePostIdLinkInSelection(
+	selection,
+	setSelection,
+	isLoadingItems,
+	items
+) {
+	const {
+		params: { postId },
+	} = useLocation();
+	const [ postIdToSelect, setPostIdToSelect ] = useState( postId );
+	useEffect( () => {
+		if ( postId ) {
+			setPostIdToSelect( postId );
+		}
+	}, [ postId ] );
```

Comment:
I'm not quite following why this extra state is needed. Could you clarify it?

Reply:
Hi @ellatrix, when we have a postId on the URL parameter we should include it in the selection. But we can not include it in the selection right way we can only include it in the selection after the data is loaded. So basically what we are doing is marking the postIdToSelect in state as a "Todo" and when items are loaded if there is a selection on "Todo" we change the selection.